### PR TITLE
Improve the performance of torch.linear()

### PR DESCRIPTION
**Pull request by razarmehr**
- Also Clean up redundant headers and namespaces from Linear.mm 

This should improve the Bert sample in #77799 by ~3x

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Linear.mm [hunk 1]
```diff
@@ -114,17 +82,26 @@ Tensor _mps_linear(
           }
           else
           {
-            MPSGraphTensor* xMulWTTensor = [mpsGraph matrixMultiplicationWithPrimaryTensor:inputTensor
+            MPSGraphTensor* inputFlattened = inputTensor;
+            bool doReshape = false;
+            // workaround to improve the performance with 3D+ inputs
+            if (input_size.size() > 2 && input_size[0] > 1 && input_size[1] >= 1 && input_size[1] <= 32) {
+              doReshape = true;
+              inputFlattened = [mpsGraph flatten2DTensor:inputTensor axis:-1 name:nil];
```
Comment (DenisVieriu97) [inline aten/src/ATen/native/mps/operations/Linear.mm:90]: does `axis:-1` here mean to flatten all the axis?

Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/Linear.mm:90]: Oh nvm, this is the same as in PyTorch

**Review by DenisVieriu97 — APPROVED** [review]
Looks good

File: aten/src/ATen/native/mps/operations/Linear.mm (hunk 1 above)
Comment (razarmehr) [reply aten/src/ATen/native/mps/operations/Linear.mm:90]: Yes.

**Pull request merged**